packages/syft/src/syft/service/api/api_service.py

```diff
@@ -491,39 +434,26 @@ def call(
         path: str,
         *args: Any,
         **kwargs: Any,
-    ) -> SyftSuccess | SyftError:
+    ) -> SyftSuccess:
         """Call a Custom API Method"""
         custom_endpoint = self.get_code(
             context=context,
             endpoint_path=path,
-        )
-        if isinstance(custom_endpoint, SyftError):
-            return custom_endpoint
-
-        exec_result = custom_endpoint.exec(context, *args, **kwargs)
-
-        if isinstance(exec_result, SyftError):
-            return Ok(exec_result)
-
+        ).unwrap()
+        
+        exec_result = custom_endpoint.exec(context, *args, **kwargs).unwrap()
         action_obj = ActionObject.from_obj(exec_result)
         action_service = cast(ActionService, context.node.get_service(ActionService))
         try:
-            result = action_service.set_result_to_store(
+            return action_service.set_result_to_store(
                 context=context,
                 result_action_object=action_obj,
                 has_result_read_permission=True,
-            )
-            if result.is_err():
-                return SyftError(
-                    message=f"Failed to set result to store: {result.err()}"
-                )
-
-            return Ok(result.ok())
+            ).unwrap()
         except Exception as e:
             # stdlib
             import traceback
-
-            return Err(value=f"Failed to run. {e}, {traceback.format_exc()}")
+            raise SyftException(public_message=f"Failed to run. {e}, {traceback.format_exc()}")
```

Comment:
wen can remove the try except here, the error handling already takes care of the stacktrace and we can just do unwrap(public_message="Failed to run")